packages/pancake-uikit/src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -24,10 +25,10 @@ const DropdownMenu: React.FC<DropdownMenuProps> = ({
   ...props
 }) => {
   const [isOpen, setIsOpen] = useState(false);
-  const [targetRef, setTargetRef] = useState<HTMLDivElement | null>(null);
-  const [tooltipRef, setTooltipRef] = useState<HTMLDivElement | null>(null);
+  const targetRef = useRef<HTMLDivElement | null>(null);
+  const tooltipRef = useRef<HTMLDivElement | null>(null);
```

Comment:
hm....
@chef-jojo no problems after switching from useState to useRef? I remember that `usePopper` specifically requires useState to track reference instead of useRef 🤔  

https://popper.js.org/react-popper/v2/
> Note: the usePopper hook intentionally takes the DOM node, not refs, in order to be able to update when the nodes change. A callback ref is used here to permit this behaviour, and useState is an appropriate way to implement this.

I don't remember the exact reason... maybe its only applicable to the elements that change position. 

This change is only for satisfying useOnClickOutside type, right? Maybe it can accept the callback ref.... or just plug in ad-hoc clickOutsideHandler right here 🤔

Reply:
Thanks for catching this, I didn't notice that
Let's stick with useState like the doc said